sway-core/src/language/ty/declaration/function.rs

```diff
@@ -182,12 +180,105 @@ impl MaterializeConstGenerics for TyFunctionDecl {
     }
 }
 
+/// Rename const generics when the name inside the struct/enum declaration  does not match
+/// the name in the impl.
+fn rename_const_generics_on_function(
```

Comment:
The renaming will not work if we just swap the const generic arguments in the impl. The both arguments will end up having the same value.

E.g., if we take the `TwoConstGenerics<T, N1, N2>` used in the tests and implement it like:

```sway
impl<T, const N2: u64, const N1: u64> TwoConstGenerics<T, N2, N1> {
  ...
}
```
and then instantiate with, e.g.:

```sway
    let e: TwoConstGenerics<u8, 1, 2> = TwoConstGenerics::<u8, 1, 2>::A([1u8]);
```
both `e.return_n1()` and `e.return_n2()` will return `2`.

Reply:
Will fix this in a future PR.